spec/scheduling-tasks.md

```diff
@@ -169,14 +183,36 @@ A <dfn>scheduler task queue</dfn> is a [=struct=] with the following [=struct/it
 </div>
 
 <div algorithm>
-  To <dfn>create a scheduler task queue</dfn> with {{TaskPriority}} |priority|:
+  To <dfn>create a scheduler task queue</dfn> with {{TaskPriority}} |priority| and |removalSteps|:
 
   1. Let |queue| be a new [=scheduler task queue=].
   1. Set |queue|'s [=scheduler task queue/priority=] to |priority|.
   1. Set |queue|'s [=scheduler task queue/tasks=] to a new empty [=set=].
+  1. Set |queue|'s [=scheduler task queue/removal steps=] to |removalSteps|.
   1. Return |queue|.
 </div>
 
+<div algorithm>
+  To <dfn>create a task handle</dfn> given a promise |result| and an {{AbortSignal}} or null
+  |signal|:
+
+  1. Let |handle| be a new [=task handle=].
+  1. Set |handle|'s [=task handle/task=] to null.
+  1. Set |handle|'s [=task handle/queue=] to null.
+  1. Set |handle|'s [=task handle/abort steps=] to the following steps:
+    1. [=Reject=] |result| with |signal|'s [=AbortSignal/abort reason=].
+    1. If |task| is not null, then
+      1. [=scheduler task queue/Remove=] |task| from |queue|.
+      1. If |queue| is [=scheduler task queue/empty=], then run |queue|'s [=scheduler task
+         queue/removal steps=].
+  1. Set |handle|'s [=task handle/task complete steps=] to the following steps:
+    1. If |signal| is not null, then [=AbortSignal/remove=] |handle|'s [=task handle/abort steps=]
+       from |signal|.
```

Comment:
from |handle|?

Reply:
Nevermind, I see now, once we complete this handle we don't need the signal to call the abort steps any more.

Reply:
That's right. I need to fix this in some other specs too, but abort algorithms can accumulate, which can be leaky.